### PR TITLE
Feature #11914: adding state management to Group model.

### PR DESCRIPTION
**Pull request by SilverYoCha**
Creating new version of SQL scripts about busCore.

Modifying GroupDAO SQL queries in order to take into account the new table fields.
Making that REMOVED state is always filtered, but not explicitly requested.

Fixing unindexation of subgroups of removed/deleted groups.

Taking into account the REMOVED State into Group synchronization.

Modifying to group page in order to wire the new remove/delete.
Adding the page to manage removed groups.

Adding the scheduler in charge of the removing of removed groups.

Linked to PR https://github.com/Silverpeas/Silverpeas-Components/pull/776

### CODE REVIEW

**Review by mmoqui — CHANGES_REQUESTED** [review]

File: core-api/src/main/java/org/silverpeas/core/admin/user/model/Group.java [hunk 1]
```diff
@@ -148,4 +151,44 @@ static Group getById(String groupId) {
...
+   */
+  Date getSaveDate();
+
+  /**
+   * Please use {@link Group#isValidState()} to retrieve group validity information.
+   * Please use {@link Group#isRemovedState()} to retrieve group removed information.
+   * This method returns the stored state information but not the functional information.
+   * @return the state of the group.
+   */
+  GroupState getState();
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-api/src/main/java/org/silverpeas/core/admin/user/model/Group.java:173]: Group#getState() allows to get the state value of the group and manipulates it to do some stuffs.
The other methods allows to check a specific state without coding the condition.

File: core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_de.properties [hunk 2]
```diff
@@ -50,7 +50,9 @@ JDP.userPanelDomain = Verwaltung der Domain
 JDP.userPanelGroup = Verwaltung der Gruppe
 
 JDP.groupAdd = Erstellen einer Gruppe
-JDP.groupDelConfirm = Sind Sie sicher, diese Gruppe l\u00f6schen zu wollen?
+JDP.groupRemConfirm=Do you really want to delete this group (and its subgroups)?
+JDP.groupDelConfirm=permanently?
+JDP.groupDelConfirmHelp=if checked, group and its subgroups are permanently deleted, otherwise they can be later restored
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_de.properties:55]: OK

File: core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_de.properties [hunk 3]
```diff
@@ -164,8 +166,14 @@ JDP.rights.assign.sourceRightsUserPanel = W\u00e4hlen Sie den Benutzer oder die
 JDP.rights.assign.nodeAssignRights = Auch zuweisen Rechten f\u00fcr Ordner der EDM
 JDP.rights.assign.MessageOk = Die Zugriffsrechte wurden assign
 JDP.rights.assign.MessageNOk = Problem bei der Vergabe von Zugriffsrechten
+JDP.removedGroupAccess=Show the removed groups
+JDP.removedGroups=removed groups
+JDP.groupRemoveDate=Date of the group remove
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_de.properties:171]: OK

File: core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_de.properties [hunk 4]
```diff
@@ -164,8 +166,14 @@ JDP.rights.assign.sourceRightsUserPanel = W\u00e4hlen Sie den Benutzer oder die
...
+JDP.groupAutomaticDeletionDate=Date of permanent deletion
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_de.properties:172]: OK

File: core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_en.properties [hunk 5]
```diff
@@ -163,6 +165,12 @@ JDP.rights.assign.sourceRightsUserPanel = Select the user or group
 JDP.rights.assign.nodeAssignRights = Also assign rights to the folders of the EDM
 JDP.rights.assign.MessageOk = The access rights have been assigned
 JDP.rights.assign.MessageNOk = Problem while assigning access rights
+JDP.removedGroupAccess=Show the removed groups
+JDP.removedGroups=removed groups
+JDP.restoreSelectionConfirmHelp=(parent groups will also be restored)
+JDP.deleteSelectionConfirmHelp=(subgroups will also be deleted)
+JDP.groupRemoveDate=Date of the group remove
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_en.properties:172]: OK

File: core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_en.properties [hunk 6]
```diff
@@ -163,6 +165,12 @@ JDP.rights.assign.sourceRightsUserPanel = Select the user or group
...
+JDP.groupAutomaticDeletionDate=Date of permanent deletion
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/properties/org/silverpeas/jobDomainPeas/multilang/jobDomainPeasBundle_en.properties:173]: OK

File: core-configuration/src/main/config/resources/StringTemplates/core/admin/domain/removedGroupHelp_en.st [hunk 7]
```diff
@@ -0,0 +1,11 @@
+List of <b>removed</b> groups.
+<p>
+At this level, it is possible to :
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/resources/StringTemplates/core/admin/domain/removedGroupHelp_en.st:3]: OK (you wanted to write ':' and not ';', that is right?)

File: core-configuration/src/main/config/resources/StringTemplates/core/admin/domain/removedGroupHelp_en.st [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+<ul>
+<li>restore groups. Their rights to which they belonged <b>at time of their remove</b> are also restored.</li>
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-configuration/src/main/config/resources/StringTemplates/core/admin/domain/removedGroupHelp_en.st:5]: OK

File: core-library/src/main/java/org/silverpeas/core/admin/service/DefaultAdministration.java [hunk 9]
```diff
@@ -499,9 +502,7 @@ private void deleteSpaceProfiles(SpaceInst spaceInst) throws AdminException {
     List<ComponentInst> components = spaceInst.getAllComponentsInst();
     for (ComponentInst component : components) {
       for (int p = 0; p < component.getNumProfileInst(); p++) {
-        if (!component.getProfileInst(p).isInherited()) {
-          deleteProfileInst(component.getProfileInst(p).getId());
-        }
+        deleteProfileInst(component.getProfileInst(p).getId());
```
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-library/src/main/java/org/silverpeas/core/admin/service/DefaultAdministration.java:505]: Mmm, I suddenly have a doubt. I check that right away and I come to validate this point.

File: core-api/src/main/java/org/silverpeas/core/admin/user/model/Group.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmoqui) [reply core-api/src/main/java/org/silverpeas/core/admin/user/model/Group.java:173]: Ok but this method doesn't seem to be used and according to the doc its handling is expected to be useless

File: core-library/src/main/java/org/silverpeas/core/admin/service/DefaultAdministration.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (SilverYoCha) [reply core-library/src/main/java/org/silverpeas/core/admin/service/DefaultAdministration.java:505]: OK, I checked again this modification.
This code is performed only into the case of space logical deletion, so it was a mistake to not delete inherited profiles too on a such treatment.
In order to clarify that, I renamed the method.

File: core-api/src/main/java/org/silverpeas/core/admin/user/model/Group.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmoqui) [reply core-api/src/main/java/org/silverpeas/core/admin/user/model/Group.java:173]: oki

**Review by mmoqui — APPROVED** [review]

**Pull request merged**